docs/design-doc-provider-identity-based-auth.md

```diff
@@ -0,0 +1,571 @@
+# Identity Based Authentication for Crossplane Providers
+
+* Owner: Alper Rifat Uluçınar (@ulucinar)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+Crossplane providers need to authenticate themselves to their respective Cloud
+providers. This establishes an identity for the Crossplane provider that's later
+used by the Cloud provider to authorize the requests made by the Crossplane
+provider and for various other purposes such as audit logging, etc. Each
+Crossplane provider supports a subset of the underlying Cloud provider's
+authentication mechanisms and this subset is currently implemented in-tree,
+i.e., in the Crossplane provider's repo, there exists a CRD that's
+conventionally named as `ProviderConfig` and each managed resource of the
+provider has a [v1.Reference][v1.reference] to a `ProviderConfig` CR. This
```

Comment:
The `[v1.Reference]` link is left as TODO and opens a 404 error page, I think it would be good to fix this.

Reply:
Unfortunately can't find document link to resolve this.

Reply:
We can use the following here: https://docs.crossplane.io/v1.12/concepts/managed-resources/#providerconfigref

Reply:
Fixed